### PR TITLE
update text detection model from DB to PPOCRv3DB

### PR DESCRIPTION
**Pull request by WanliZhong**
This PR is follow https://github.com/opencv/opencv_zoo/pull/158
This PR should be merged after 4.8.0 released

TODO list:
- [x] update model
- [x] quantize model (Accuracy drop because the models are sensitive to quantization)
- [x] update cpp demo after https://github.com/opencv/opencv_zoo/pull/176
- [x] update benchmark data

### CODE REVIEW

**Comment by fengyuentau** [comment]
Friendly reminder.

File: benchmark/README.md [hunk 1]
```diff
@@ -280,8 +288,10 @@ mean       median     min        input size   model
 153.89     153.96     153.43     [224, 224]   MPPersonDet with ['person_detection_mediapipe_2023mar.onnx']
 44.29      44.03      43.62      [128, 256]   YoutuReID with ['person_reid_youtu_2021nov.onnx']
 91.28      92.89      89.79      [256, 256]   MPPose with ['pose_estimation_mediapipe_2023mar.onnx']
-254.78     256.13     245.60     [640, 480]   DB with ['text_detection_DB_IC15_resnet18_2021sep.onnx']
-254.98     255.20     245.60     [640, 480]   DB with ['text_detection_DB_TD500_resnet18_2021sep.onnx']
+427.53     428.67     425.63     [640, 480]   PPOCRv3DB with ['text_detection_ch_ppocrv3_2023may.onnx']
+427.79     429.28     425.63     [640, 480]   PPOCRv3DB with ['text_detection_en_ppocrv3_2023may.onnx']
+414.07     429.46     387.26     [640, 480]   PPOCRv3DB with ['text_detection_ch_ppocrv3_2023may_int8.onnx']
+406.10     407.83     383.41     [640, 480]   PPOCRv3DB with ['text_detection_en_ppocrv3_2023may_int8.onnx']
```
Comment (WanliZhong) [inline benchmark/README.md:294]: All the backends are faster with the new model, only FP16 is slower than before.

File: benchmark/README.md [hunk 2]
```diff
@@ -443,8 +459,6 @@ mean       median     min        input size   model
 5.17       5.26       5.09       [192, 192]   MPPalmDet with ['palm_detection_mediapipe_2023feb.onnx']
 16.45      16.44      16.31      [224, 224]   MPPersonDet with ['person_detection_mediapipe_2023mar.onnx']
 5.58       5.57       5.54       [128, 256]   YoutuReID with ['person_reid_youtu_2021nov.onnx']
-17.15      17.18      16.83      [640, 480]   DB with ['text_detection_DB_IC15_resnet18_2021sep.onnx']
-17.95      18.61      16.83      [640, 480]   DB with ['text_detection_DB_TD500_resnet18_2021sep.onnx']
```
Comment (WanliZhong) [inline benchmark/README.md:447]: The CANN backend doesn't support `HardSwish` node then causes fallback to CPU. Related PR: https://github.com/opencv/opencv/issues/24154

**Comment by fengyuentau** [comment]
I propose to:

1. resolve conflicts,
2. add this model to models/text_detection_ppocrv3,
3. delete text_detection_db in antoher pr.

**Comment by fengyuentau** [comment]
Hardswish has been supported for CANN backend. Could you update benchmark results on Ascend NPUs?

**Comment by WanliZhong** [comment]
Before OpenCV 4.9 release, I will try to upgrade PPOCR to version 4

**Review by fengyuentau — APPROVED** [review]
LGTM 👍

**Pull request merged**